File: Keshav Banka/Day 9/9A.cpp
class Solution {
public:
    int reverse(int x) {
        int temp = x, res = 0;
        while (temp) {
            int rem = temp % 10;
            if (res > INT_MAX / 10 || res < INT_MIN / 10) return 0;
            res *= 10;
            res += rem;
            temp /= 10;
        }
        return res;
    }
};

File: Mahek Jhamb/Day 9/9A.py
class Solution:
    def reverse(self, x: int) -> int:
        rev = 0
        sign = 1 if x >= 0 else -1
        x *= sign
        while x > 0:
            rev = rev * 10 + x % 10
            x //= 10
        rev *= sign
        return rev if rev >= -(2**31) and rev <= (2**31 - 1) else 0